packages/flutter/lib/src/rendering/editable.dart

```diff
@@ -349,26 +349,28 @@ class RenderEditable extends RenderBox {
       ..isTextField = true;
 
     if (_selection?.isValid == true) {
-      if (_textPainter.getOffsetBefore(_selection.extentOffset) != null) {
-        config.addAction(SemanticsAction.moveCursorBackwardByCharacter, () {
-          final int offset = _textPainter.getOffsetBefore(_selection.extentOffset);
+      if (_textPainter.getOffsetAfter(_selection.extentOffset) != null) {
```

Comment:
tiny nit: Swapping the two conditionals at this level would make the diffs simpler to read...I can't tell if it would change the algorithm or not though.

Reply:
Done.